packages/core-contracts/contracts/mocks/proxy/ForwardingProxyMock.sol

```diff
@@ -0,0 +1,38 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "../../proxy/ForwardingProxy.sol";
+
+contract ForwardingProxyMock is ForwardingProxy {
+    bytes32 private _slot0;
```

Comment:
This mock stores the implementation in a very primitive form: just use the nth slot.

The `ForwardingProxy` just forwards calls, it does not determine how it gets or upgrades its implementation, only that its somewhere in storage.